### PR TITLE
Automate employment of standby in case of failure

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

File: deploy/vm/ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ssh_private_key_file: the ssh key for the VM configuration
+ssh_public_key_file: the ssh key for the VM configuration
+vm_user: user to be created for the VMs
+allowed_source_ip_prefix: The source ip addresses that are allowed to access (Access Control List)
+n: number of worker VMs to create
+m: number of standby VMs to create
+az_domain_name: prefix to use for the domain name of the VMs. The domain name can not have upper case or special characters except '-'
+
+##Faileover case: 1) Migrate the disks from the failed VM to a standby VM. 2) Move the standby VM behind the load balancer the failed VM is attached to
+Command: ansible-playbook migrate_stanby_to_worker.yml --extra-vars="{resource_group: "resource_group", migrate_to_vm: "name-standby-vm", decommission_vm: "name-failed-hana-vm" }"
```
Comment (tniek) [inline deploy/vm/ansible/README.md]: nit: standby (with a d)

File: deploy/vm/ansible/migrate_stanby_to_worker.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+#This playbook employs a standby node as a worker node. The steps performed are as follows:
```
Comment (tniek) [inline deploy/vm/ansible/migrate_stanby_to_worker.yml]: "standby node as a worker node"... doesn't really make sense.
It should read "uses a standby VM to configure it as worker node in HANA"

File: deploy/vm/ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+resource_group: name of the azure resource group to be created
+ssh_private_key_file: the ssh key for the VM configuration
+ssh_public_key_file: the ssh key for the VM configuration
+vm_user: user to be created for the VMs
+allowed_source_ip_prefix: The source ip addresses that are allowed to access (Access Control List)
+n: number of worker VMs to create
+m: number of standby VMs to create
+az_domain_name: prefix to use for the domain name of the VMs. The domain name can not have upper case or special characters except '-'
+
+##Faileover case: 1) Migrate the disks from the failed VM to a standby VM. 2) Move the standby VM behind the load balancer the failed VM is attached to
```
Comment (tniek) [inline deploy/vm/ansible/README.md]: What about vIP migration?

Comment (pabowers) [inline deploy/vm/ansible/README.md]: typo: should be "Failover"

File: deploy/vm/ansible/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#n+m scale out scenario
+
+##How to deploy n+m scale out HANA cluster:
```
Comment (pabowers) [inline deploy/vm/ansible/README.md]: These lines aren't showing up as being formatted correctly on Github although I know it should be valid Markdown

File: deploy/vm/ansible/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Example:
+ansible-playbook create_n_m_hana_cluster.yml --extra-vars="{resource_group: "resourcegroupname", ssh_private_key_file: "/tmp/sshkey", ssh_public_key_file: "/tmp/sshkey.pub", vm_user: "azureuser", allowed_source_ip_prefix: "X.X.X.X/24", n: 2, m:0, az_domain_name: "mydomain"}"
```
Comment (pabowers) [inline deploy/vm/ansible/README.md]: nit: put in a code block

File: deploy/vm/ansible/migrate_stanby_to_worker.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          enable_accelerated_networking: True
+          ip_configurations:
+              - name: hana-ip-config
+                public_ip_address_name: "{{ standby_nic.ansible_facts.azure_networkinterfaces[0].properties.ipConfigurations[0].properties.publicIPAddress.id.split('/')[-1] }}"
+                primary: yes
+                load_balancer_backend_address_pools:
+                    - name: "{{ nic_res.ansible_facts.azure_networkinterfaces[0].properties.ipConfigurations[0].properties.loadBalancerBackendAddressPools[0].id.split('/')[-1] }}"
+                      load_balancer: "{{ nic_res.ansible_facts.azure_networkinterfaces[0].properties.ipConfigurations[0].properties.loadBalancerBackendAddressPools[0].id.split('/')[-3] }}"
+#
+#          register: nic_sample
```
Comment (pabowers) [inline deploy/vm/ansible/migrate_stanby_to_worker.yml]: nit: remove commented out code

File: deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+#Following is the format in which the disks are passed to this role
```
Comment (pabowers) [inline deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml]: nit: space between "#" and Following

File: deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#    "diskSizeGB": 256,
+#    "lun": 2,
+#    "managedDisk": {
+#       "id": "/subscriptions/XXX/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Compute/disks/hanaXdata",
+#       "storageAccountType": "Premium_LRS"
+#     },
+#     "name": "hana1data"
+#   }
+#   ]
+     - name: unmount the managed disk to the VM
```
Comment (pabowers) [inline deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml]: nit: capitalize name

File: deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+       azure_rm_managed_disk:
+          name: "{{ item.name }}"
+          resource_group: "{{ az_resource_group }}"
+          disk_size_gb: "{{ item.diskSizeGB}}"
+          storage_account_type: Premium_LRS
+       with_items: "{{ az_managed_disk }}"
+     - name: remount the managed disk to another VM
```
Comment (pabowers) [inline deploy/vm/ansible/roles/migrate-hana-disk/tasks/main.yml]: nit: Capitalize name

File: deploy/vm/ansible/README.md (hunk 3 above)
Comment (persiaAziz) [reply deploy/vm/ansible/README.md]: I am moving the standby VM behind the load balancer. If vIP is the IP of the load balancer, then it should not need any change or migration, so there is a seamless change behind the loadbalancer. Am I right?

File: deploy/vm/ansible/README.md (hunk 4 above)
Comment (persiaAziz) [reply deploy/vm/ansible/README.md]: huh, actually it does not :/ let me see what is happening

**Pull request merged**